src/Cli/dotnet/commands/dotnet-tool/install/ToolInstallLocalCommand.cs

```diff
@@ -48,12 +51,70 @@ internal class ToolInstallLocalCommand : CommandBase
             _toolManifestEditor = toolManifestEditor ?? new ToolManifestEditor();
             _localToolsResolverCache = localToolsResolverCache ?? new LocalToolsResolverCache();
             _toolLocalPackageInstaller = new ToolInstallLocalInstaller(parseResult, toolPackageInstaller);
+            _allowPackageDowngrade = parseResult.GetValue(ToolInstallCommandParser.AllowPackageDowngradeOption);
         }
 
         public override int Execute()
         {
             FilePath manifestFile = GetManifestFilePath();
-            return Install(manifestFile);
+            var existingPackageWithPackageId = _toolManifestFinder.Find(manifestFile).Where(p => p.PackageId.Equals(_packageId));
+
+            if (!existingPackageWithPackageId.Any())
+            {
+                return Install(manifestFile);
+            }
+
+            var existingPackage = existingPackageWithPackageId.Single();
+            var toolDownloadedPackage = _toolLocalPackageInstaller.Install(manifestFile);
+
+            InstallLogic(existingPackage, toolDownloadedPackage, manifestFile);
+            return 0;
+        }
+
+        public int InstallLogic(ToolManifestPackage existingPackage, IToolPackage toolDownloadedPackage, FilePath manifestFile)
```

Comment:
I'd suggest renaming this to `InstallToolUpdate` or something similar to make it clearer what it does.